test/shared/src/main/scala/zio/test/mock/TestRandom.scala

```diff
@@ -93,6 +117,9 @@ final case class TestRandom(ref: Ref[TestRandom.Data]) extends Random.Service[An
 object TestRandom {
   val DefaultData: Data = Data()
 
+  def apply(data: Data): UIO[TestRandom] =
```

Comment:
These are super handy! I would call them `make`, though, instead of `apply`, because they build it inside an effect, rather than return it directly.